showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -73,10 +104,25 @@ class ShowkaseProcessor @JvmOverloads constructor(
         logger.publishMessages(messager)
     }
 
-    private fun processComponentAnnotation(roundEnvironment: XRoundEnv): Set<ShowkaseMetadata.Component> {
+    private fun processComponentAnnotation(
+        roundEnvironment: XRoundEnv,
+    ): Set<ShowkaseMetadata.Component> {
         val showkaseComposablesMetadata = processShowkaseAnnotation(roundEnvironment)
         val previewComposablesMetadata = processPreviewAnnotation(roundEnvironment)
-        return (showkaseComposablesMetadata + previewComposablesMetadata)
+
+        // This is for getting custom annotations from class path
+        val customAnnotationMetadataFromClassPath =
```

Comment:
Why are we picking from the classpath here? Why not do it at the same place where we do it for `@Preview` and `@ShowkaseComposable` annotations

Reply:
I think the reason I have to split this up into two methods, is that the `processCustomAnnotation` method only uses the supported annotations type qualified name and was meant for KAPT to work. But the `processCustomAnnotationFromClassPath` was meant to get the stored annotation metadata from each round and based on that check if we had any qualified names we could search for. The reason we have to split them up is that we are using some of the other properties from the stored metadata that we store on each round I believe.  Maybe I can remove the `processCustomAnnotation` and merge that into the `processCustomAnnotationFromClassPath` but I think that might risk it becoming very messy? And I kinda don't know how to handle that we get preview name and such from the stored metadata and not from the supported compile option name thingy 🤔